frontend/packages/core/src/dialog.tsx

```diff
@@ -1,27 +1,61 @@
-import React from "react";
-import MuiDialog from "@material-ui/core/Dialog";
-import DialogActions from "@material-ui/core/DialogActions";
-import DialogContent from "@material-ui/core/DialogContent";
-import DialogTitle from "@material-ui/core/DialogTitle";
-import Typography from "@material-ui/core/Typography";
+import * as React from "react";
+import styled from "@emotion/styled";
+import type { DialogProps as MuiDialogProps } from "@material-ui/core";
+import {
+  Dialog as MuiDialog,
+  DialogActions as MuiDialogActions,
+  DialogContent as MuiDialogContent,
+  DialogTitle as MuiDialogTitle,
+  IconButton as MuiIconButton,
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
 
-export interface DialogProps {
+const DialogTitle = styled(MuiDialogTitle)({
+  display: "flex",
+  justifyContent: "space-between",
+  fontSize: "16px",
+  padding: "16px 16px 0 16px",
+  fontWeight: 400,
+  color: "#000000",
+});
+
+const IconButton = styled(MuiIconButton)({
+  height: "16px",
+  width: "16px",
+  padding: "0",
+  color: "#000000",
+});
+
+const DialogContent = styled(MuiDialogContent)({
+  padding: "8px 16px",
+  fontSize: "12px",
+  fontWeight: 400,
+  color: "rgba(13, 16, 48, 0.6)",
+});
+
+const DialogActions = styled(MuiDialogActions)({
+  borderTop: "1px solid rgba(13, 16, 48, 0.12)",
+  padding: "0 8px",
+});
+
+export interface DialogProps extends Pick<MuiDialogProps, "open"> {
   title: string;
-  content: string;
-  open: boolean;
+  children: React.ReactNode;
+  actions: React.ReactNode;
   onClose: () => void;
```

Comment:
i'm not sure that onClose should ever do anything other than dismiss the dialog, that seems like a bad UX pattern to allow devs to tie behavior to onClose. any use case in mind?